### ISSUE TITLE
Use a markup language that allows for comments in //flutter/ci/builders/*

### ISSUE THREAD

**Issue by chinmaygarde** [flutter/flutter#156493]
We use JSON today. But the manifests are fairly large and hard to parse. As a relative newbie to tinkering on these, I would like to add additional inline documentation via comments. But JSON makes this hard to do.

Don't have a particularly strong opinion on a specific markup language, just want to express a frustration with JSON as we use it today.

**Comment by christopherfujino** [comment]
Agreed. I hate YAML too, but I think it would be the best replacement format, as it would be trivial to convert to json (I suspect JSON was implemented because the recipes can only natively parse either JSON or protobufs).

**Comment by chinmaygarde** [comment]
@matanlurey Suggested adding an object like so:

```json
"_comment": "The comment goes here...",
```

I'm going to start doing that in the meantime.

@gaaclarke also suggested JSON5(?).

**Comment by christopherfujino** [comment]
> @gaaclarke also suggested JSON5(?).

I don't have a strong opinion, as long as I'm not the one maintaining the dart parser :)
